llama.cpp

```diff
@@ -1577,7 +1581,7 @@ static bool llama_eval_internal(
 
             // projection (no bias)
             cur = ggml_mul_mat(ctx0,
-                    model.layers[il].wo,
+                    layer.wo,
```

Comment:
There is no `layer` in scope at this point. Did you actually try to compile your changes locally?

Creating a variable for the current layer to avoid having to do `model.layers[il]` isn't really going to make any measurable performance difference (so it's not "optimizing" anything) but it may make the code more readable. Also, it doesn't make sense to only change some of `model.layers[il]` to `layers` but leave a bunch of others in the same function. There's probably less of an argument for the `.at()` change in the other function.

Anyway, I have no authority to tell you to do anything but my suggestion is that before you create a PR you at least compile and test your changes locally and ensure that 1) the project can still compile and 2) your changes don't break existing functionality.

Reply:
perhaps this problem appeared after resolving conflicts that caused confusion, I can remove commit changes that eliminate the repetition layer objects